node/src/chain/fsm.rs

```diff
@@ -190,6 +191,7 @@ impl<N: Network, DB: database::DB, VM: vm::VMExecution> SimpleFSM<N, DB, VM> {
                         next.on_entering(&b, peer_addr).await;
                         self.curr = State::OutOfSync(next);
                     }
+                    // block not accepted, should we return?
```

Comment:
I think we should, if we state that `on_block_event` returns the block only if it's accepted. If we don't return here, we would return `Ok(block)` at the end.